arrow-core-test/src/main/kotlin/arrow/test/laws/FoldableLaws.kt

```diff
@@ -114,4 +130,31 @@ object FoldableLaws {
         foldM.equalUnderTheLaw(foldL, EQ)
       }
     }
+
+  fun <F> Foldable<F>.firstOrNoneReturnsNoneIfEmpty(G: Gen<Kind<F, Int>>) =
+    forAll(G) { fa: Kind<F, Int> ->
+      if (fa.isEmpty()) fa.firstOrNone().isEmpty()
+      else fa.firstOrNone().isDefined()
+    }
+
+  fun <F> Foldable<F>.firstOrNoneReturnsNoneIfPredicateFails(G: Gen<Kind<F, Int>>) =
+    forAll(G) { fa: Kind<F, Int> ->
+      fa.firstOrNone { false }.isEmpty()
+    }
+
+  fun <F> Foldable<F>.`firstOrNone is consistent with find`(G: Gen<Kind<F, Int>>, EQ: Eq<Option<Int>>) =
+    forAll(G) { fa: Kind<F, Int> ->
+      fa.firstOrNone().equalUnderTheLaw(fa.find { true }, EQ)
+    }
+
+  fun <F> Foldable<F>.`firstOrNone is consistent with find predicate`(G: Gen<Kind<F, Int>>, EQ: Eq<Option<Int>>) =
+    forAll(Gen.intPredicate(), G) { f: (Int) -> Boolean, fa: Kind<F, Int> ->
+      fa.firstOrNone(f).equalUnderTheLaw(fa.find { f(it) }, EQ)
+    }
+
+  fun <F> Foldable<F>.`toList turn items into a list`(G: Gen<Kind<F, Int>>, EQ: Eq<Int>) =
+    forAll(G) { fa: Kind<F, Int> ->
+      val expected = fa.foldRight(Eval.now(emptyList<Int>())) { v, acc -> acc.map { listOf(v) + it } }.value()
+      fa.toList().size.equalUnderTheLaw(expected.size, EQ)
+    }
```

Comment:
Just check both lists if they are equal. Only checking the size might not uncover ordering problem for example.

Reply:
I was struggling to get `Eq<List<Int>>`

Reply:
Two choices:
- Use `ListK.eq` and lift using `k`
- Use `Eq.any` both are trivial types hence `==` works just fine